design-proposals/klip-17-sql-union.md

```diff
@@ -0,0 +1,193 @@
+This KLIP is related because it removes `INSERT INTO` which inserts into an existing stream.
+* [KLIP-20](klip-20_remove_terminate.md): Remove `TERMINATE` statement: which requires the (1 - 0..1)
+relationship between data source and persistent query that this KLIP will restore.
+* [KLIP-21](klip-21_correct_insert_values_semantics.md): Correct 'INSERT VALUES' semantics: which 
+depends on [KLIP-19](klip-19-materialize-views.md).
+
+We propose removing `INSERT INTO` functionality and replacing it with support for the standard SQL 
+`UNION ALL` operator. 
+
+The standard SQL `UNION` operator is used to combine the result-set of two or more `SELECT` statement, 
+with certain limitations:
+
+* Each `SELECT` statement within the `UNION` must have the same number of columns
+* The columns must also have similar data types
+* The columns in each SELECT statement must also be in the same order
+
+The `UNION` operator selects only distinct rows by default, where as `UNION ALL` allows duplicates.
+We are looking to support _merging_ multiple streams, which, aside from the impracticalities of de-duppling, 
+actually requires that duplicates are included. Hence this KLIP proposes the introduction of `UNION ALL`, 
+not `UNION`.
+
+UNION ALL Syntax:
+
+```sql
+SELECT column_name(s) FROM stream1
+UNION ALL
+SELECT column_name(s) FROM stream2;
+```
+
+Additional sources can be merged by adding additional `UNION ALL` operators:
+
+```sql
+SELECT column_name(s) FROM stream1
+UNION ALL
+SELECT column_name(s) FROM stream2
+UNION ALL
+SELECT column_name(s) FROM stream3;
+```
+
+Note: The column names in the result are equal to the column names in the first `SELECT` statement in the union.
+
+Switching to `UNION ALL` operator will enable users to merge multiple streams together without the drawbacks
+of `INSERT INTO`:
+
+* It allows two or more streams to be merge together in a single statement,
+* which is a simple conceptual model for users.
+* Under the hood it will use a single KS topology,
+* meaning only a single copy of reference table state-stores
+* and only a single query to terminate before the sink stream can be dropped, (or non at all with 
+[KLIP-20](klip-20_remove_terminate.md)).
+
+## What is in scope
+
+* Removal of `INSERT INTO` statements from the syntax and code. (Breaking change).
+* Introduction of `UNION ALL` support to merge multiple streams.
+* Arbitrary `SELECT` statements in the union, as long as they result in a stream, not a table.
+* Support for unions in CSAS and transient push queries.
+
+## What is not in scope
+
+* Introduction of `UNION` operator: which removes duplicates from the result set.
+We may choose to introduce this later, but is not required in any MVP.
+* Introduction of `UNION ALL` support on tables. though it may not be much work to add this later. 
+Support for tables is not required to remove `INSERT INTO` and unblock the other KLIPs, 
+hence it has been de-scoped.
```

Comment:
ksqlDB requires that table always have a primary key -- hence, I don't think it will be possible to add `UNION ALL` for tables. (Just a side remark.)